dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/route/RouteService.java

```diff
@@ -188,26 +185,25 @@ public Route getRouteWithDecryptedAuth(@Nonnull String id) {
   public ResponseEntity<String> execute(
       Route route, UserDetails userDetails, Optional<String> subPath, HttpServletRequest request)
       throws IOException, BadRequestException {
-    HttpHeaders headers = filterRequestHeaders(request);
 
+    HttpHeaders headers = filterRequestHeaders(request);
     route.getHeaders().forEach(headers::add);
-
     addForwardedUserHeader(userDetails, headers);
 
+    MultiValueMap<String, String> queryParameters = new LinkedMultiValueMap<>();
```

Comment:
I think we want to avoid using any other collections and utils than JDK provided ones. The stream API and methods like `Map.computeIfAbsent` can be used to do the same. 
```java
map.computeIfAbsent("key", k -> new ArrayList<>()).addAll(...)
```